firedrake/checkpointing.py

```diff
@@ -707,6 +712,82 @@ def _save_mesh_topology(self, tmesh):
             perm_is.setName(None)
             self.viewer.popGroup()
 
+    @PETSc.Log.EventDecorator("GetTimesteps")
+    def get_timestepping_history(self, mesh, name):
+        """
+        Retrieve the timestepping history and indices for a specified function within a mesh.
+
+        This method is primarily used in checkpointing scenarios during timestepping simulations.
+        It returns the indices associated with each function call in the timestepping mode,
```

Comment:
```suggestion
        It returns the indices associated with each function stored in the timestepping mode,
```
Is this what you meant?